applications/ChimeraApplication/chimera_application.h

```diff
@@ -0,0 +1,240 @@
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_application.h"
+#include "includes/variables.h"
+
+
+
+namespace Kratos {
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+class KratosChimeraApplication : public KratosApplication {
+public:
+	///@name Type Definitions
+	///@{
+
+
+	/// Pointer definition of KratosChimeraApplication
+	KRATOS_CLASS_POINTER_DEFINITION(KratosChimeraApplication);
+
+	///@}
+	///@name Life Cycle
+	///@{
+
+	/// Default constructor.
+	KratosChimeraApplication();
+
+	/// Destructor.
+	~KratosChimeraApplication() override = default;
+
+
+	///@}
+	///@name Operators
+	///@{
+
+
+	///@}
+	///@name Operations
+	///@{
+
+	virtual void Register() override;
+
+
+
+	///@}
+	///@name Access
+	///@{
+
+
+	///@}
+	///@name Inquiry
+	///@{
+
+
+	///@}
+	///@name Input and output
+	///@{
+
+	/// Turn back information as a string.
+	virtual std::string Info() const override{
+		return "KratosChimeraApplication";
+	}
+
+	/// Print information about this object.
+	virtual void PrintInfo(std::ostream& rOStream) const override{
+		rOStream << Info();
+		PrintData(rOStream);
+	}
+
+	///// Print object's data.
+	virtual void PrintData(std::ostream& rOStream) const override{
+  		KRATOS_WATCH("in my application");
+  		KRATOS_WATCH(KratosComponents<VariableData>::GetComponents().size() );
+
+		rOStream << "Variables:" << std::endl;
+		KratosComponents<VariableData>().PrintData(rOStream);
+		rOStream << std::endl;
+		rOStream << "Elements:" << std::endl;
+		KratosComponents<Element>().PrintData(rOStream);
+		rOStream << std::endl;
+		rOStream << "Conditions:" << std::endl;
+		KratosComponents<Condition>().PrintData(rOStream);
+    }
+
+
+	///@}
+	///@name Friends
+	///@{
+
+
+	///@}
+
+protected:
+	///@name Protected static Member Variables
+	///@{
+
+
+	///@}
+	///@name Protected member Variables
+	///@{
+
+
+	///@}
+	///@name Protected Operators
+	///@{
+
+
+	///@}
+	///@name Protected Operations
+	///@{
+
+
+	///@}
+	///@name Protected  Access
+	///@{
+
+
+	///@}
+	///@name Protected Inquiry
+	///@{
+
+
+	///@}
+	///@name Protected LifeCycle
+	///@{
+
+
+	///@}
+
+private:
+	///@name Static Member Variables
+	///@{
+
+	// static const ApplicationCondition  msApplicationCondition;
+
+	///@}
+	///@name Member Variables
+	///@{
+
+	///@}
+	///@name Private Operators
+	///@{
+
+
+	///@}
+	///@name Private Operations
+	///@{
+
+
+	///@}
+	///@name Private  Access
+	///@{
+
+
+	///@}
+	///@name Private Inquiry
+	///@{
+
+
+	///@}
+	///@name Un accessible methods
+	///@{
+
+	/// Assignment operator.
+	KratosChimeraApplication& operator=(KratosChimeraApplication const& rOther);
+
+	/// Copy constructor.
+	KratosChimeraApplication(KratosChimeraApplication const& rOther);
```

Comment:
I'd explicitly `delete` the assignment operator and the copy constructor (although the behaviour is the same). This also allows to make the file shorter by removing the protected and private sections.

Reply:
Lets also change this in the application generator then ?

Reply:
Because this is the default file from application generator.